langserver/references.go

```diff
@@ -75,6 +74,12 @@ func (h *LangHandler) handleTextDocumentReferences(ctx context.Context, conn jso
 		return PathHasPrefix(path, h.init.RootImportPath)
 	}
 
+	// findRefCtx is used in the findReferences function. It has its own
+	// context so we can stop finding references once we have reached our
+	// limit.
+	findRefCtx, stop := context.WithCancel(ctx)
```

Comment:
we only want to cancel finding references. There are other operations happening, but it would actually work if we just shadowed at the beginning. However, in reality all we want to do is cancel reference finding, so I believe this more accurately shows the intent + future proof for extra things we may add.